bach/bach/series/series.py

```diff
@@ -390,6 +391,59 @@ def copy_override(self,
             index_sorting=self._index_sorting if index_sorting is None else index_sorting
         )
 
+    def unstack(self,
+                level: int = -1,
+                fill_value: Optional[Union[int, float, str, UUID]] = None,
+                aggregation: str = 'max'):
+        """
+        Pivot a level of the index labels.
+
+        Returns a(n unsorted) DataFrame with the values of the unstacked index as columns. In case of
+        duplicate index values that are unstacked, `aggregation` is used to aggregate the values.
+
+        :param level: selects the level of the index that is unstacked. Currently only -1 supported.
+        :param fill_value: replace missing values resulting from unstacking. Should be of same type as the
+            series that is unstacked.
+        :param aggregation: method of aggregation, in case of duplicate index values. Supported are 'max' and
+            'min'
+        :returns: DataFrame
+        """
+        index_dict = self.index
+        if len(index_dict) <= 1:
+            raise NotImplementedError('index must be a multi level index to unstack')
+        if level != -1:
+            raise NotImplementedError('only last index can be unstacked')
+        if aggregation not in ['max', 'min']:
+            raise ValueError(f"aggregation is '{aggregation}', should be 'min' or 'max'")
+
+        name_index_last, series_last = index_dict.popitem()
+        values = series_last.unique().values
+        if None in values or np.nan in values:
+            raise ValueError("index contains empty values, cannot be unstacked")
+        name_series = self.name
+        remaining_indexes = list(index_dict.keys())
+        df = self.to_frame().reset_index().groupby(remaining_indexes)  # type: ignore
```

Comment:
```suggestion
        df = self.to_frame()
        df.reset_index(inplace=True)
        df = df.groupby(remaining_indexes)
```
mypy complains because `reset_index` has `Optional` typing due to `inplace=True` case